### ISSUE TITLE
Strictly return a word with given length

### ISSUE THREAD

**Issue by Shinigami92** [faker-js/faker#721]
### Clear and concise description of the problem

https://github.com/faker-js/faker/pull/714#discussion_r837124211

MO it feels a bit weird if a word with given length is not found then use random length 🤔
IMO we should change behavior for all such cases to a min/max length and search between that. If nothing found return undefined or throw an error?

### Suggested solution

We need to discuss whether we should return undefined, null or an error in such a case

**Comment by pkuczynski** [comment]
I think for some values it might be hard to keep the exact length (like when `length > 10`). So maybe we should simply call this param `maxLength`?

**Comment by Shinigami92** [comment]
It's not that simple, because then users would loose current behavior to generate an exact length word
So we need a minLength and a maxLength, but I also think we should do this via an options param, so maybe `{ minLength, maxLength }` or `length: { min, max }`

**Comment by pkuczynski** [comment]
Well, the current behavior is actually `maxLength` and not `length`. The param name is misleading :)

**Comment by pkuczynski** [comment]
I agree those functions could have 2 signatures:

```
xyz (maxLength: number)
xyz (options: { minLength?: number, maxLength?: number} )
```

**Comment by Shinigami92** [comment]
> Well, the current behavior is actually `maxLength` and not `length`. The param name is misleading :)


No it's not. The current behavior is: if length match was found use exact length, if not fallback to random length.

**Comment by pkuczynski** [comment]
Which means that current `length` actually describes `maxLength`, no? Or at least kind of. More like `desiredLength` ;)

**Comment by ST-DDT** [comment]
As the word functions currently use desiredLength we should keep this as the default. Maybe adding other variants as an option.

I suggest this signature
`word( length?: number | { min?: number, max?: number, fallbackToAnyLength: boolean = true})`

That way it would be non breaking.

**Comment by xDivisionByZerox** [comment]
This got solved with #1407.
